### PR TITLE
Task5/import to s3

### PR DESCRIPTION
**Pull request by kirill-ianchuk**
### Task 5.1
https://4xh4vm4dbf.execute-api.eu-west-1.amazonaws.com/dev/import?name=products.csv

Frontend app:
https://d2958odpxdpwzu.cloudfront.net/admin/products
PR: https://github.com/kirill-ianchuk/nodejs-aws-fe/pull/4

### Task 5.2
Implemented importFileParser function

Cloud Watch logs:
image

### Additional tasks

- [x] async/await is used in lambda functions
- [x] importProductsFile lambda is covered by unit tests
- [x] At the end of the stream the lambda function moves the file from the uploaded folder into the parsed folder

### Expected total score: 8

### CODE REVIEW

**Review by RidgeA — APPROVED** [review]
LGTM

File: import-service/src/handlers/importFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    };
+
+    const results = [];
+
+    s3.getObject(params)
+        .createReadStream()
+        .on('error', reject)
+        .pipe(csv())
+        .on('error', reject)
+        .on('data', (data) => results.push(data))
```
Comment (numatay) [inline import-service/src/handlers/importFileParser.ts]: I think storing chunks of streamed data in the array is not a good idea. Using streams helps us to process data bits by bits, so the memory footprint of the app would be tiny. But your implementation defeats its purpose by storing the whole file in memory. Imagine, for example, processing a CSV file over 3 gigabytes, this lambda function would be terminated due to memory overflow.

https://aws.amazon.com/about-aws/whats-new/2017/11/aws-lambda-doubles-maximum-memory-capacity-for-lambda-functions/#:~:text=AWS%20Lambda%20Doubles%20Maximum%20Memory%20Capacity%20for%20Lambda%20Functions,-Posted%20On%3A%20Nov&text=You%20can%20now%20allocate%203008MB,to%20your%20functions%20was%201536MB.

**Comment by numatay** [comment]
Well done. 8 pts.

**Comment by kirill-ianchuk** [comment]
@numatay Thanks for your comment regarding stream, I've updated the code to log each record during stream processing

**Pull request merged**